src/function.rs

```diff
@@ -385,6 +408,60 @@ where
         let db = unsafe { self.view_caster().downcast_unchecked(db) };
         self.accumulated_map(db, key_index)
     }
+
+    fn is_serializable(&self) -> bool {
+        C::SERIALIZABLE
+    }
+
+    fn should_serialize(&self, zalsa: &Zalsa) -> bool {
+        if !C::SERIALIZABLE {
+            return false;
+        }
+
+        // We only serialize the query if there are any memos associated with it.
+        for instance_key in <C::SalsaStruct<'_> as SalsaStructInDb>::instances(zalsa) {
+            let memo_ingredient_index = self
+                .memo_ingredient_indices
+                .get(instance_key.ingredient_index());
+
+            let memo = self.get_memo_from_table_for(
+                zalsa,
+                instance_key.key_index(),
+                memo_ingredient_index,
+            );
+
+            if memo.is_some_and(|memo| memo.value.is_some()) {
```

Comment:
We should probably also skip over any memos that aren't finalized (it was built as part of some fix point query but isn't part of the final result). You can detect them with `maybe_provisional(...) == true` or `validate_may_be_provisional(...) == false` (with `allow_non_finalized` set to `false`). 

A second and related question is whether we should persist memos that haven't been verified in the most recent revision. This would allow us to prune old query results but it has the downside that we might end up pruning results only because they weren't queried in a while. I'm okay deferring this to later but it might be worth adding a TODO comment here.